src/supervisor/watchers/handlers/index.ts

```diff
@@ -58,97 +85,133 @@ import { RETRYABLE_NETWORK_ERRORS } from '../types';
+    clusterEndpoint: '/apis/apps/v1/daemonsets',
+    namespacedEndpoint: '/apis/apps/v1/watch/namespaces/{namespace}/daemonsets',
     handlers: {
       [DELETE]: daemonSetWatchHandler,
     },
-    listFactory: (namespace) => () => paginatedDaemonSetList(namespace),
+    clusterListFactory: () => () => paginatedClusterDaemonSetList(),
+    namespacedListFactory: (namespace) => () =>
+      paginatedNamespacedDaemonSetList(namespace),
   },
   [WorkloadKind.Deployment]: {
-    endpoint: '/apis/apps/v1/watch/namespaces/{namespace}/deployments',
+    clusterEndpoint: '/apis/apps/v1/deployments',
+    namespacedEndpoint:
+      '/apis/apps/v1/watch/namespaces/{namespace}/deployments',
     handlers: {
       [DELETE]: deploymentWatchHandler,
     },
-    listFactory: (namespace) => () => paginatedDeploymentList(namespace),
+    clusterListFactory: () => () => paginatedClusterDeploymentList(),
+    namespacedListFactory: (namespace) => () =>
+      paginatedNamespacedDeploymentList(namespace),
   },
   [WorkloadKind.ReplicaSet]: {
-    endpoint: '/apis/apps/v1/watch/namespaces/{namespace}/replicasets',
+    clusterEndpoint: '/apis/apps/v1/replicasets',
+    namespacedEndpoint:
+      '/apis/apps/v1/watch/namespaces/{namespace}/replicasets',
     handlers: {
       [DELETE]: replicaSetWatchHandler,
     },
-    listFactory: (namespace) => () => paginatedReplicaSetList(namespace),
+    clusterListFactory: () => () => paginatedClusterReplicaSetList(),
+    namespacedListFactory: (namespace) => () =>
+      paginatedNamespacedReplicaSetList(namespace),
   },
   [WorkloadKind.StatefulSet]: {
-    endpoint: '/apis/apps/v1/watch/namespaces/{namespace}/statefulsets',
+    clusterEndpoint: '/apis/apps/v1/statefulsets',
+    namespacedEndpoint:
+      '/apis/apps/v1/watch/namespaces/{namespace}/statefulsets',
     handlers: {
       [DELETE]: statefulSetWatchHandler,
     },
-    listFactory: (namespace) => () => paginatedStatefulSetList(namespace),
+    clusterListFactory: () => () => paginatedClusterStatefulSetList(),
+    namespacedListFactory: (namespace) => () =>
+      paginatedNamespacedStatefulSetList(namespace),
   },
   [WorkloadKind.DeploymentConfig]: {
+    clusterEndpoint: '/apis/apps.openshift.io/v1/deploymentconfigs',
     /** https://docs.openshift.com/container-platform/4.7/rest_api/workloads_apis/deploymentconfig-apps-openshift-io-v1.html */
-    endpoint:
+    namespacedEndpoint:
       '/apis/apps.openshift.io/v1/watch/namespaces/{namespace}/deploymentconfigs',
     handlers: {
       [DELETE]: deploymentConfigWatchHandler,
     },
-    listFactory: (namespace) => () => paginatedDeploymentConfigList(namespace),
+    clusterListFactory: () => () => paginatedClusterDeploymentConfigList(),
+    namespacedListFactory: (namespace) => () =>
+      paginatedNamespacedDeploymentConfigList(namespace),
   },
 };
 
-async function isSupportedWorkload(
+export const WATCH_WHOLE_CLUSTER = '';
```

Comment:
I don't understand what this const does?

Reply:
I guess to decide if the whole cluster watch is set up ... is empty string the safest way to do this?
```suggestion
export const WATCH_WHOLE_CLUSTER = 'WATCH_WHOLE_CLUSTER';
```